public/app/features/alerting/unified/components/receivers/PayloadEditor.tsx

```diff
@@ -0,0 +1,102 @@
+import { css } from '@emotion/css';
+import React, { useState } from 'react';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { Stack } from '@grafana/experimental';
+import { Button, Icon, TextArea, Tooltip, useStyles2 } from '@grafana/ui';
+import { TestTemplateAlert } from 'app/plugins/datasource/alertmanager/types';
+
+import { GenerateAlertDataModal } from './form/GenerateAlertDataModal';
+
+export const RESET_TO_DEFAULT = 'Reset to default';
+
+export function PayloadEditor({
+  payload,
+  setPayload,
+  defaultPayload,
+}: {
+  payload: string;
+  defaultPayload: string;
+  setPayload: React.Dispatch<React.SetStateAction<string>>;
+}) {
+  const styles = useStyles2(getStyles);
+  const onReset = () => {
+    setPayload(defaultPayload);
+  };
+
+  const [isEditingAlertData, setIsEditingAlertData] = useState(false);
+
+  const onCloseEditAlertModal = () => {
+    setIsEditingAlertData(false);
+  };
+
+  const onOpenEditAlertModal = () => setIsEditingAlertData(true);
+
+  const onAddAlertList = (alerts: TestTemplateAlert[]) => {
+    onCloseEditAlertModal();
+    setPayload((payload) => {
+      const payloadObj = JSON.parse(payload);
```

Comment:
This needs some additional error handling, if we try to add an alert to a malformed JSON payload this will crash the UI I think

Reply:
trueeee

Reply:
I solved this issue this way 👀 @gillesdemey 

https://user-images.githubusercontent.com/33540275/234650891-91e85880-c3ed-4600-9476-2e54274674fc.mp4

Reply:
perfect!